httptransport/client/differ.go

```diff
@@ -0,0 +1,176 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"path"
+	"strings"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/quay/claircore/libvuln/driver"
+
+	"github.com/quay/clair/v4/httptransport"
+	"github.com/quay/clair/v4/matcher"
+)
+
+var _ matcher.Differ = (HTTP)(nil)
+
+// DeleteUpdateOperations attempts to delete the referenced update operations.
+func (c HTTP) DeleteUpdateOperations(ctx context.Context, ref ...uuid.UUID) error {
+	u, err := c.addr.Parse(httptransport.UpdatesAPIPath)
+	if err != nil {
+		return err
+	}
+
+	// Spawn a few requests that will write their result into "errs".
+	//
+	// These'll most likely be multiplexed and to the same host, so pick a nice
+	// lowish number like 4.
+	//
+	// Don't use an errgroup because we want to actually issue all the DELETEs,
+	// not stop all requests on the first error.
+	var wg sync.WaitGroup
+	item := make(chan int)
+	errs := make([]error, len(ref))
+	for i := 0; i < 4; i++ {
```

Comment:
I think I'm a little confused about how this method works. You limit the amount of deletes that can be performed by 4, but do not return any indication to the client if they send more then four refs into the method, that any refs >= 5 will not be handled.

if that is the case, I think we either want to do one of two things:
- place the wait operations inside a for loop, doing "batches" of deletes with a safe concurrency bound
- return an error to the client to attempt delete again with less refs. 

lmk if I'm off here.

Reply:
This for loop spawns for 4 concurrent workers that pull off the `item` channel, then the for loop below sends all the indexes down that channel.